### PR TITLE
Fix: Don't `.forget()` RAF closure

### PR DESCRIPTION
**Pull request by jprochazk**
The closure is now stored in `WebRunner` and dropped in the next `request_animation_frame` instead of being "leaked" via `forget`

### CODE REVIEW

File: crates/eframe/src/web/web_runner.rs [hunk 1]
```diff
@@ -193,20 +191,47 @@ impl WebRunner {
...
+            return Ok(());
+        }
+
         let window = web_sys::window().unwrap();
         let closure = Closure::once({
             let runner_ref = self.clone();
-            move || events::paint_and_schedule(&runner_ref)
+            move || {
+                // we can paint now, so clear the animation frame
+                // this drop the `closure` and allows another
```
Comment (emilk) [inline crates/eframe/src/web/web_runner.rs:204]:
```suggestion
                // We can paint now, so clear the animation frame.
                // This drops the `closure` and allows another
```

I think?

File: crates/eframe/src/web/web_runner.rs [hunk 2]
```diff
@@ -193,20 +191,47 @@ impl WebRunner {
...
+                // animation frame to be scheduled
+                let _ = runner_ref.frame.take();
```
Comment (emilk) [inline crates/eframe/src/web/web_runner.rs:206]:
```suggestion
                runner_ref.frame = None;
```

File: crates/eframe/src/web/web_runner.rs [hunk 3]
```diff
@@ -115,9 +112,10 @@ impl WebRunner {
     pub fn destroy(&self) {
         self.unsubscribe_from_all_events();
 
-        if let Some(id) = self.request_animation_frame_id.get() {
+        if let Some(frame) = self.frame.take() {
             let window = web_sys::window().unwrap();
-            window.cancel_animation_frame(id).ok();
+            window.cancel_animation_frame(frame.id).ok();
+            drop(frame.closure);
```
Comment (emilk) [inline crates/eframe/src/web/web_runner.rs:118]: this happens implicitly anyway

```suggestion
```

**Review by emilk — APPROVED** [review]
Thanks for breaking this out!

File: crates/eframe/src/web/web_runner.rs (hunk 2 above)
Comment (jprochazk) [reply crates/eframe/src/web/web_runner.rs]: This doesn't work because it's an `Rc<RefCell<Option<...>>>`

**Pull request merged**